libraries/AP_RangeFinder/AP_RangeFinder_uLanding.cpp

```diff
@@ -55,25 +55,31 @@ bool AP_RangeFinder_uLanding::get_reading(uint16_t &reading_cm)
     // read any available lines from the uLanding
     float sum = 0;
     uint16_t count = 0;
-    uint8_t  index = 0;
+    bool hdr_found = false;
 
     int16_t nbytes = uart->available();
     while (nbytes-- > 0) {
         uint8_t c = uart->read();
-        // ok, we have located start byte
-        if (c == 72 && index == 0) {
+        
+        if (c == ULANDING_HDR && !hdr_found) {
+            // located header byte
             linebuf_len = 0;
-            index       = 1;
+            hdr_found   = true;
         }
-        // now it is ready to decode index information
-        if (index == 1) {
-            linebuf[linebuf_len] = c;
-            linebuf_len ++;
-            if (linebuf_len == 3) {
-                index = 0;
-                sum += (linebuf[2]&0x7F) *128 + (linebuf[1]&0x7F);
+        // decode index information
+        if (hdr_found) {
+            linebuf[linebuf_len++] = c;
+
+            if (linebuf_len == 6) {
+            // received six bytes of data
+                if (((linebuf[1] + linebuf[2] + linebuf[3] + linebuf[4]) & 0xFF) == linebuf[5]) {
+                // checksum passed
```

Comment:
Nitpicking: can we move these two comments to either:

- be above the *if*s, **or**
- indent them as they should inside the *if*